.checkpatch.conf

```diff
@@ -1,7 +1,7 @@
 --emacs
 --summary-file
 --show-types
---max-line-length=80
+--max-line-length=100
```

Comment:
this needs to be discussed, I am fine with that. I also think we should decide if we keep following the linux style *with* changes or if we should establish our own based a list of rules (from Linux and others) and grow from there. Changes that happen to Linux should not be automatically adopted into Zephyr, thats why we have a few exceptions already.

Reply:
I remember seeing the LKML discussion about this and as someone who uses a text editor with multiple columns I would not be thrilled with this change to the coding style

Reply:
If this is changed, will .clang-format also be updated to assist with code formatting?

Reply:
If accepted, we need to update `.editorconfig`to match.

Reply:
I remember thinking about this, but apparently didn't do it.  Done now: all line length checks updated in one combined commit.